integration_tests/tests/integration_test.rs

```diff
@@ -236,8 +329,107 @@ fn test_echo_with_large_array_input() -> Result<()> {
 
     let large_array: Vec<i32> = (0..=u16::MAX as usize).map(|x| x as i32).collect();
     assert_eq!(
-        run_example_with_input_and_msgpack_output("echo", serde_json::json!(large_array))?,
+        run_wasm_api_example("echo", serde_json::json!(large_array))?,
         serde_json::json!(large_array)
     );
     Ok(())
 }
+
+#[test]
+fn test_fuel_consumption_within_threshold() -> Result<()> {
+    BENCHMARK_EXAMPLE_RESULT
+        .as_ref()
+        .map_err(|e| anyhow::anyhow!("Failed to prepare example: {}", e))?;
+    let input = generate_cart_with_size(2, true);
+    let wasm_api_input = prepare_wasm_api_input(input.clone())?;
+    let (_, wasm_api_fuel) = run_example("cart-checkout-validation-wasm-api", wasm_api_input)?;
+    eprintln!("WASM API fuel: {}", wasm_api_fuel);
+    // Using a target fuel value as reference similar to the Javy example
+    assert_fuel_consumed_within_threshold(16522, wasm_api_fuel);
+    Ok(())
+}
+
+#[test]
+fn test_benchmark_comparison_with_input() -> Result<()> {
+    BENCHMARK_EXAMPLE_RESULT
+        .as_ref()
+        .map_err(|e| anyhow::anyhow!("Failed to prepare example: {}", e))?;
+    BENCHMARK_NON_WASM_API_EXAMPLE_RESULT
+        .as_ref()
+        .map_err(|e| anyhow::anyhow!("Failed to prepare non-WASM API example: {}", e))?;
+
+    let input = generate_cart_with_size(2, true);
+
+    let wasm_api_input = prepare_wasm_api_input(input.clone())?;
+    let (wasm_api_output, wasm_api_fuel) =
+        run_example("cart-checkout-validation-wasm-api", wasm_api_input)?;
+    let wasm_api_value = decode_msgpack_output(wasm_api_output)?;
+
+    let wasi_json_input = prepare_wasi_json_input(input)?;
+    let (non_wasm_api_output, non_wasm_api_fuel) =
+        run_example("cart-checkout-validation-wasi-json", wasi_json_input)?;
+    let non_wasm_api_value = decode_json_output(non_wasm_api_output)?;
+
+    assert_eq!(wasm_api_value, non_wasm_api_value);
+    assert!(
+        wasm_api_fuel < non_wasm_api_fuel,
+        "WASM API fuel usage ({}) should be less than non-WASM API fuel usage ({}) when early exit is possible",
```

Comment:
The Wasm API one isn't early exiting in this case, right? Should we update this error message?

Reply:
Updated. this  message is better for future proofing too